lib/src/rive_core/state_machine_controller.dart

```diff
@@ -418,7 +429,7 @@ class StateMachineController extends RiveAnimationController<CoreContext> {
           in nestedArtboard.animations.whereType<NestedStateMachine>()) {
         switch (hitEvent) {
           case ListenerType.down:
-            nestedStateMachine.pointerDown(nestedPosition);
+            nestedStateMachine.pointerDown(nestedPosition, event!);
```

Comment:
Should we not make this propagate hit test too? Something like:
```
if(nestedStateMachine.pointerDown(nestedPosition, event!)) {
  foundTarget = true;
}
```
For cases where all pointer events are handled in nested state machines...or does it not matter because they eventually call into their own state machine controllers (with their own ImmediateMultiDragGestureRecognizers).

What happens if both this StateMachineController and a nested one calls addPointer to their recognizers?

Reply:
I first did that, but then I thought every `StateMachineController` can report itself if `HitShape` were hit. Not sure how the framework will handle it, but all controllers would handle the event, since they don't care about flutter's gesture disambiguation. What do you think?

Reply:
Yeah I think it should work out since they all have their individual controllers, but I'm not familiar with this API. Maybe we should do a test with this @zplata (use some file that has Rive Listeners in the main state machine and in nested state machines too).

Reply:
@luigi-rosso Tested on a local example I had - the pointer down listener for both the parent and nested artboard seemed to take fine, with these changes!